app/adapters/osu_mirrors/__init__.py

```diff
@@ -112,7 +152,120 @@ async def fetch_beatmap_zip_data(beatmapset_id: int) -> bytes | None:
             "mirror_weight": mirror.weight,
             "beatmapset_id": beatmapset_id,
             "ms_elapsed": ms_elapsed,
-            "data_size": len(beatmap_zip_data) if beatmap_zip_data is not None else 0,
+            "data_size": len(mirror_response.data) if mirror_response.data else 0,
+        },
+    )
+    return mirror_response.data
+
+
+async def fetch_beatmap_background_image(beatmap_id: int) -> bytes | None:
```

Comment:
A lot of the code in these can be extracted out into a common "do request to mirrors" base function

Reply:
cba to do it now; but can come sometime in the future